lib/symengine.rb

```diff
@@ -33,6 +33,33 @@ def Function(n)
     def evalf(operand, prec=53, real=false)
         return _evalf(operand, prec, real)
     end
+    def lambdify(exp, syms)
+      eval(SymEngine::Utils::lambdify_code(exp, syms))
+    end
+  end
+  module Utils
+    class << self
+      def evalf_dirichlet_eta(exp)
+        SymEngine::evalf(SymEngine::dirichlet_eta(exp))
+      end
+      def evalf_zeta(exp)
+          SymEngine::evalf(SymEngine::zeta(exp))
+      end
+      def lambdify_code(exp, syms)
+        str = exp.to_s  
+        sym_map = syms.join(",")
+        str.gsub!(/[\d\.]+/, 'Rational(\0,1)')
+        replacements = { sin:"Math.sin", cos: "Math.cos", tan: "Math.tan",
+                         asin:"Math.asin", acos: "Math.acos", atan: "Math.atan",
+                         sinh:"Math.sinh", cosh: "Math.cosh", tanh: "Math.tanh",
+                         asinh:"Math.asinh", acosh: "Math.acosh", atanh: "Math.atanh", 
+                         pi: "Math::PI", E: "Math::E",
+                         dirichlet_eta: "SymEngine::Utils::evalf_dirichlet_eta",
+                         zeta: "SymEngine::Utils::evalf_zeta", gamma: "Math.gamma" }
+        replacements.each {|key, value| str.gsub!(/(\b#{key}\b)/, value)}
```

Comment:
Better to use `inject` here instead of modyfying gsub:
```ruby
str = replacements.inject(str) { |res, (from, to)| res.gsub(/(\b#{from}\b)/, to)}
```

Overall, modern ruby tends to more "functional" style, without modyfying variables inplace.

Another thing I'm concerned a bit is ineffectiveness of constructing ton of regexes on the fly. Maybe replacements hash could be defined somewhat like:
```ruby
REPLACEMENTS = {
  asin:"Math.asin", acos: "Math.acos", ....
}.map { |from, to| [/(\b#{from}\b)/, to] }.to_h.freeze
```